From eb9f72e79d870cc25140c8d7ea5f379b0a7bcde8 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Fri, 10 Nov 2017 17:01:12 +0800
Subject: [PATCH 2790/5242] MLK-16804-07 ARM64: dts: Add cpuidle state info
 node for imx8mq

commit  342c1efafa53b452380303def058895493547c45 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add cpuidle states node for imx8mq.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Reviewed-by: Anson Huang <anson.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi |   37 +++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 52fdd82..b162af2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -42,6 +42,30 @@
 		gpio4 = &gpio5;
 	};
 
+	cpus {
+		idle-states {
+			entry-method = "psci";
+
+			CPU_SLEEP: cpu-sleep {
+				compatible = "arm,idle-state";
+				arm,psci-suspend-param = <0x0000000>;
+				entry-latency-us = <700>;
+				exit-latency-us = <250>;
+				min-residency-us = <1000>;
+			};
+
+			CLUSTER_SLEEP: cluster-sleep {
+				compatible = "arm,idle-state";
+				arm,psci-suspend-param = <0x1010033>;
+				local-timer-stop;
+				entry-latency-us = <1000>;
+				exit-latency-us = <700>;
+				min-residency-us = <2700>;
+				wakeup-latency-us = <1500>;
+			};
+		};
+	};
+
 	memory@40000000 {
 		device_type = "memory";
 		reg = <0x00000000 0x40000000 0 0xc0000000>;
@@ -1161,6 +1185,7 @@
 };
 
 &A53_0 {
+	cpu-idle-states = <&CPU_SLEEP &CLUSTER_SLEEP>;
 	operating-points = <
 		/* kHz    uV */
 		1000000 900000
@@ -1174,3 +1199,15 @@
 	clock-latency = <61036>;
 	#cooling-cells = <2>;
 };
+
+&A53_1 {
+	cpu-idle-states = <&CPU_SLEEP &CLUSTER_SLEEP>;
+};
+
+&A53_2 {
+	cpu-idle-states = <&CPU_SLEEP &CLUSTER_SLEEP>;
+};
+
+&A53_3 {
+	cpu-idle-states = <&CPU_SLEEP &CLUSTER_SLEEP>;
+};
-- 
1.7.9.5

